R/ResultFlagsIndependent.R

```diff
@@ -1254,3 +1254,307 @@ TADA_FindPotentialDuplicatesSingleOrg <- function(.data) {
   .data <- TADA_OrderCols(.data)
   return(.data)
 }
+
+#' TADA.DepthCategory.Flag
+#'
+#' Creates a new column, TADA.DepthCategory.Flag with values: "No
+#' depth info", "Epilimnion-surface", "Hypolimnion-bottom", and
+#' "Metalimnion/Thermocline-middle". Categories can be defined to start as:
+#' less than 2m depth = "Epilimnion", from bottom up to 2m from
+#' bottom= "Hypolimnion-bottom", and the full depth profile for
+#' "Metalimnion/Thermocline-middle".
+#'
+#' When more than one result is available for a MonitoringLocationIdentifier,
+#' ActivityStartDate, OrganizationIdentifier, and TADA.CharacteristicName,
+#' user input defines which value (average, max, or min value) to use for that
+#' day and location. User input also defines whether aggregation occurs over
+#' each depth category or for the entire depth profile.
+#'
+#' @param .data TADA dataframe which must include the columns ADD DEPTH COLUMNS
+#'
+#' @param daily_agg Character argument; with options "no", "ave", "min", or
+#' "max". The default is daily_agg = "no". When daily_agg = "no", all results
+#' will be retained. When daily_agg == "ave", the mean value in each group of
+#' results (as determined by the by category param) will be determined for each
+#' MonitoringLocation, ActivityDate, Organization ID, and TADA.CharacteristicName combination.
+#' When daily_agg == "min" or when daily_agg == "max", the min or max
+#' value in each group of results (as determined by the by category param) will
+#' be determined for each MonitoringLocation, ActivityDate, and TADA.CharacteristicName
+#' combination. An additional column, TADA.ResultValueAggregation.DepthProfile.Flag will be added
+#' to describe aggregation.
+#'
+#' @param bycategory Boolean argument with options "TRUE" or "FALSE". The default is
+#' by category = "FALSE" which means that any descriptive statistics calculated are
+#' based on the entire water column at a Monitoring Location. When bycategory = "TRUE",
+#' any calculated descriptive statistics are determined for each depth category for each
+#' Monitoring Location.
+#'
+#' @param clean Boolean argument with options "TRUE" or "FALSE". The default is
```

Comment:
"clean" or "aggregatedonly"? 

We use clean in other other function to remove data that is invalid or not applicable to the dataset. In this situation for depth data, clean might be more applicable to use to remove results that do not include depth data (I wonder if that should be it's own separate function - a flag function for finding results that do not include any depth information). 

aggregatedonly is similar to other flag functions flaggedonly function input.

Reply:
Thanks for the explanation. I will change clean to "aggregatedonly" for this function. I think a "clean" option for this function also makes sense.